### PR TITLE
Google search console v2

### PR DESCRIPTION
**Pull request by ishachopra0401**
## What do these changes do?

Created Google search console v2 doc
Preview URL: https://rudderstack-docs-git-google-search-console-v2-rudder-stack.vercel.app/sources/extract/google-search-console-v2/

### CODE REVIEW

File: docs/sources/extract/google-search-console-v2.mdx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+### Selecting the data to import
+
+You can choose the Google Search Console data you want to ingest by selecting the required resources:
+
+<img src="../../assets/cloud-extract-sources/gcs-resources.png" alt="Selecting the data to import" />
+
+The below table mentions the syncs and API endpoints supported by these resources from Google Search Console to your warehouse destination:
+
+| Resource | Full Refresh sync | Incremental sync | Google Search Console API endpoint |
```
Comment (ishachopra0401) [inline docs/sources/extract/google-search-console-v2.mdx:95]: @debanjan97 Can we also specify the Primary key in the table? Reference doc: https://www.rudderstack.com/docs/sources/extract/amplitude/#selecting-the-data-to-import

Comment (debanjan97) [reply docs/sources/extract/google-search-console-v2.mdx:95]: There is no field defined which will be used as a primary key. 
Rudderstack will add a new column which will act as an id.

**Review by debanjan97 — APPROVED** [review]
LGTM!

File: docs/sources/extract/google-search-console-v2.mdx [hunk 2]
```diff
@@ -0,0 +1,126 @@
+---
+title: "Google Search Console V2"
+description: Step-by-step guide to ingest your Google Search Console data into RudderStack.
+---
```
Comment (ameypv-rudder) [inline docs/sources/extract/google-search-console-v2.mdx:4]:
```suggestion
aliases: ["Google Search Console source"]
---
```

File: docs/sources/extract/google-search-console-v2.mdx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+2. Go to **Sources** > **New source** > **Cloud Extract** and select **Google Search Console V2** from the list of sources.
+3. Assign a name to your source and click **Continue**.
+
+### Connection settings
+
+Next, configure the following dashboard settings:
+
+- **Authentication Type**: Choose one of the following authentication types:
+
+    - **Oauth**: Authenticate your Google Search Console account via Google (OAuth) by clicking
```
Comment (ameypv-rudder) [inline docs/sources/extract/google-search-console-v2.mdx:35]:
```suggestion
    - **OAuth**: Authenticate your Google Search Console account via Google (OAuth) by clicking
```

File: docs/sources/extract/google-search-console-v2.mdx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    **Sign in with Google** and granting the required permissions.
```
Comment (ameypv-rudder) [inline docs/sources/extract/google-search-console-v2.mdx:36]:
```suggestion
```

File: docs/sources/extract/google-search-console-v2.mdx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+    <img src="../../assets/cloud-extract-sources/gsc-oauth-settings.png" alt="Google Analytics credentials" />
+
+    - **Service Account Key Authentication**: Enter the service account key in JSON format in the **Service Account JSON Key** field. Refer to the [Google documentation](https://cloud.google.com/iam/docs/creating-managing-service-account-keys#creating_service_account_keys) for more information on obtaining the service account key. Also, enter the email associated with your service account in the **Admin Email** field.
```
Comment (ameypv-rudder) [inline docs/sources/extract/google-search-console-v2.mdx:40]: @ishachopra0401 should we present the settings as sub-bullets, like we do usually?

**Review by ameypv-rudder — APPROVED** [review]

**Pull request merged**